test/e2e/kogitoapp_test.go

```diff
@@ -56,74 +54,108 @@ func TestKogitoApp(t *testing.T) {
 	if err != nil {
 		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
 	}
-	// more clusters can be added
-	t.Run("kogitoapp", func(t *testing.T) {
-		t.Run("Cluster1", KogitoAppCluster)
-	})
+
+	// Specifies what tests should be executed
+	tests := util.GetEnv("TESTS", "full")
+	if tests == "jvm" {
+		// Run just JVM tests
+		t.Run("kogitoapp", func(t *testing.T) {
+			t.Run("QuarkusJvm", deployKogitoQuarkusJvmExample)
+		})
+	} else if tests == "full" {
+		// Run whole test suite
+		t.Run("kogitoapp", func(t *testing.T) {
+			t.Run("QuarkusJvm", deployKogitoQuarkusJvmExample)
+			t.Run("QuarkusNative", deployKogitoQuarkusNativeExample)
+		})
+	} else {
+		log.Fatalf("wrong value of tests parameter: %v", tests)
+	}
+}
+
+func deployKogitoQuarkusJvmExample(t *testing.T) {
+	t.Parallel()
+
+	// initialize framework
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+	// get global framework variables
+	f := framework.Global
+
+	initializeKogitoOperator(t, f, ctx)
+
+	gitProjectURI := "https://github.com/kiegroup/kogito-examples"
+	contextDir := "drools-quarkus-example"
+	appName := "example-quarkus-jvm"
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		log.Fatalf("could not get namespace: %v", err)
+	}
+
+	kogitoService := getKogitoServiceStub(appName, namespace)
+	kogitoService.Spec.Build.GitSource.URI = &gitProjectURI
+	kogitoService.Spec.Build.GitSource.ContextDir = contextDir
+	kogitoService.Spec.Build.Native = false
```

Comment:
For both functions `deployKogitoQuarkusJvmExample` and `deployKogitoQuarkusNativeExample`, I think the testing logic is  the same. The only difference is the testing data, i.e. `kogitoService`. 

We can create a separate function for the testing logic, and let `deployKogitoQuarkusJvmExample` and `deployKogitoQuarkusNativeExample` create testing data and invoke the new function with different testing data.